controllers/barbican_controller.go

```diff
@@ -1185,3 +1186,18 @@ func (r *BarbicanReconciler) ensureDB(
 	instance.Status.Conditions.MarkTrue(condition.DBReadyCondition, condition.DBReadyMessage)
 	return db, ctrlResult, nil
 }
+
+func cleanupOldDeployment(ctx context.Context, c client.Client, namespace, oldName string) error {
+	oldDeployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      oldName,
+			Namespace: namespace,
+		},
+	}
+
```

Comment:
one thing we probably should do here is to validate that the corresponding barbican CR is the owner of the old deployments before delete it. in theory there could be barbican deployments from different owner CRs. similar to https://github.com/openstack-k8s-operators/openstack-operator/blob/main/pkg/openstack/galera.go#L55 . just a safety mechanism to not delete something the one being reconciled does not own.